src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -99,7 +99,10 @@ def __call__(self, organization, event):
                                 author_email = user.email
                                 gh_username_cache[gh_username] = author_email
                                 if commit_author is not None:
-                                    commit_author.update(email=author_email)
+                                    commit_author.update(
+                                        email=author_email,
+                                        external_id=gh_username,
```

Comment:
Let's flip it. Say I got the username in our private Phabricator as `matt`, then on GitHub `matt` is definitely not me and refers to someone else. I just don't see the harm in adding a prefix for the provider to just avoid the collision in the future. It'd save a migration in the future if we hit this problem.